pkg/workflowengine/impl/workflow_builder.go

```diff
@@ -0,0 +1,45 @@
+package impl
+
+import (
+	"github.com/flyteorg/flyteadmin/pkg/workflowengine/interfaces"
+
+	"github.com/flyteorg/flytestdlib/promutils"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/core"
+	"github.com/flyteorg/flytepropeller/pkg/apis/flyteworkflow/v1alpha1"
+	"github.com/flyteorg/flytepropeller/pkg/compiler/transformers/k8s"
+)
+
+type builderMetrics struct {
+	Scope                promutils.Scope
+	WorkflowBuildSuccess prometheus.Counter
+	WorkflowBuildFailure prometheus.Counter
+	InvalidExecutionID   prometheus.Counter
+}
+
+type flyteWorkflowBuilder struct {
+	metrics builderMetrics
+}
+
+func (b *flyteWorkflowBuilder) Build(
+	wfClosure *core.CompiledWorkflowClosure, inputs *core.LiteralMap, executionID *core.WorkflowExecutionIdentifier,
+	namespace string) (*v1alpha1.FlyteWorkflow, error) {
+	return k8s.BuildFlyteWorkflow(wfClosure, inputs, executionID, namespace)
```

Comment:
Should this be called out as an interface BuildFlyteWorkflow and then be able to subsitute mock or other implmentations.
The interface can be part flyteWorkflowBuilder

Reply:
I think we already have an interface, can we use that here .

Reply:
Sorry, I'm not following - this is already an implementation of the Build interface